src/userservice/tests/test_userservice.py

```diff
@@ -0,0 +1,242 @@
+        # create example user request
+        example_user_request['username'] = 'foo'
+        # send request to test client
+        response = self.test_app.post('/users', data=example_user_request)
+        # assert 409 response code
+        self.assertEqual(response.status_code, 409)
+        # assert we get correct error message
+        self.assertEqual(
+            response.json['msg'], 'user {} already exists'.format(example_user_request['username'])
+        )
+
+    def test_create_user_sql_error_500_status_code_error_message(self):
+        """test creating a new user but throws SQL error when trying to add"""
+        # mock return value of get_user which checks if user exists as None
+        self.mocked_db.return_value.get_user.return_value = None
+        # mock return value of add_user to throw SQLAlchemyError
+        self.mocked_db.return_value.add_user.side_effect = SQLAlchemyError()
+        # create example user request
+        example_user = EXAMPLE_USER_REQUEST.copy()
+        example_user['username'] = 'foo'
+        # send request to test client
+        response = self.test_app.post('/users', data=example_user)
+        # assert 500 response code
+        self.assertEqual(response.status_code, 500)
+        # assert we get correct error message
+        self.assertEqual(response.json['msg'], 'failed to create user')
+
+    def test_create_user_malformed_400_status_code_error_message(self):
+        """test creating a new user without required keys"""
+        # test each expected field missing from user request
+        for expected_field in EXPECTED_FIELDS:
+            # create example user request
+            example_user = EXAMPLE_USER_REQUEST.copy()
+            # remove a required field
+            example_user.pop(expected_field)
+            # send request to test client
+            response = self.test_app.post('/users', data=example_user)
+            # assert 400 response code
+            self.assertEqual(response.status_code, 400)
+            # assert we get correct error message
+            self.assertEqual(response.json['msg'], 'missing required field(s)')
+
+    def test_create_user_malformed_empty_400_status_code_error_message(self):
+        """test creating a new user with empty value for required key"""
+        # create example user request
+        example_user = EXAMPLE_USER_REQUEST.copy()
+        # set empty value for required key
+        example_user['username'] = ''
+        # send request to test client
+        response = self.test_app.post('/users', data=example_user)
+        # assert 400 response code
+        self.assertEqual(response.status_code, 400)
+        # assert we get correct error message
+        self.assertEqual(response.json['msg'], 'missing value for input field(s)')
+
+    def test_create_user_mismatch_password_400_status_code_error_message(self):
+        """test creating a new user with empty value for required key"""
```

Comment:
nit: think this should be `test creating a new user with mismatched password values`

Reply:
Good catch, fixed